### DISCUSSION TITLE
Creating and write to an Output window

### DISCUSSION THREAD

**Discussion by cjdutoit** [VsixCommunity/Community.VisualStudio.Toolkit#363] [Q&A] [answered]
Looking for an example of how to create and write to an Output window with the Community Toolkit from a Command

I have a command that does a few things, and I would like to write some updates to a custom command window.

```
        protected override async Task ExecuteAsync(OleMenuCmdEventArgs e)
        {
                ...
                string message = DoSomething();

                await UseOutputWindowAsync(message).ConfigureAwait(false);
        }

        private IVsOutputWindowPane _pane;
        private async Task UseOutputWindowAsync(string message)
        {
            if (_pane == null)
            {
                await JoinableTaskFactory.SwitchToMainThreadAsync(DisposalToken);
                var ow = await GetServiceAsync(typeof(SVsOutputWindow)) as IVsOutputWindow;
                Assumes.Present(ow);

                var guid = Guid.NewGuid();
                ow.CreatePane(ref guid, "My output", 1, 1);
                ow.GetPane(ref guid, out _pane);

                _pane.Activate();
                _pane.OutputStringThreadSafe($"{message}\r\n");
            }
            else
            {
                _pane.OutputStringThreadSafe($"{message}\r\n");
            }
        }

```

The code does not currently compile.  I get:
Error	CS0103	The name 'GetServiceAsync' does not exist in the current context
Error	CS0103	The name 'DisposalToken' does not exist in the current context

Any help to get past this will be greatly appreciated.  Thanks

**Comment by reduckted** [comment, marked as answer]
Here's an example of creating and writing to an output window form a command:
https://github.com/VsixCommunity/Community.VisualStudio.Toolkit/blob/5e13b112b7ab917ccc2154cdbdab32a0963bdd61/demo/VSSDK.TestExtension/Commands/ListReferencesCommand.cs

**Reply by cjdutoit** [reply]
Perfect! Thank you!
